graphs/graph_manager.py

```
@@ -0,0 +1,210 @@
+                        continue
+                        
+                    logging.debug("Generating %s", graph_type)
+                    graph_instance.data = graph_data[graph_type]
+                    file_path = await graph_instance.generate(data_fetcher)
+                    
+                    if file_path:
+                        graph_files.append((graph_type, file_path))
+                        logging.debug("Generated %s: %s", graph_type, file_path)
+                except (GraphFactoryError, IOError) as e:
+                    logging.error("Error generating %s: %s", graph_type, str(e))
+                    raise GraphGenerationError(f"Failed to generate {graph_type}: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)